torch/csrc/jit/tensorexpr/operators/misc.cpp

```diff
@@ -642,6 +642,97 @@ Tensor computeCat(
       });
 }
 
+static bool checkStackInputShape(const std::vector<BufHandle>& bufList) {
+  if (bufList.size() == 0) {
+    throw std::runtime_error("Empty input list is passed to aten::stack");
+  }
+
+  auto buf0 = bufList.at(0);
+  TORCH_INTERNAL_ASSERT(
+      buf0.node()->dims().size() > 0, buildErrorMessage("Invalid buf rank"));
+  // Check if any of the dims is 0
+  bool hasEmptyDims = false;
+  std::vector<ExprHandle> dims;
+  for (const auto& dim : buf0.dims()) {
+    if (dim.AsNode<LongImm>() && immediateAs<int64_t>(dim) == 0ll) {
+      hasEmptyDims = true;
+    }
+    dims.push_back(IRSimplifier::simplify(dim));
+  }
+
+  // Check if all input bufs have the same size
+  for (int i = 1; i < bufList.size(); i++) {
+    auto buf = bufList.at(i);
+    TORCH_INTERNAL_ASSERT(
+        buf.node()->dims().size() == dims.size(),
+        buildErrorMessage("aten::stack inputs have different sizes"));
+    auto dims_to_cmp = buf.dims();
+    for (int k = 0; k < dims.size(); k++) {
+      auto diff = IRSimplifier::simplify(
+          alloc<Sub>(dims_to_cmp.at(k).node(), dims.at(k).node()));
+      TORCH_INTERNAL_ASSERT(
+          diff->isConstant() && immediateAs<int>(diff) == 0,
+          buildErrorMessage("different aten::stack inputs dims"));
+    }
+  }
+
+  return hasEmptyDims;
+}
+
+Tensor computeStack(
+    const std::vector<ArgValue>& inputs,
+    const std::vector<ExprHandle>& outputShape,
+    const c10::optional<ScalarType>& outputType,
+    at::Device device) {
+  // NOLINTNEXTLINE(performance-unnecessary-copy-initialization)
+  auto inputList = c10::get<BufList>(inputs[0]);
+  auto argDim = inputs[1];
+  auto hasEmptyInputs = checkStackInputShape(inputList);
+  ScalarType highType = inputList[0].dtype().scalar_type();
+  for (const auto& input : inputList) {
+    auto maybe_dtype = input.dtype().scalar_type();
+    highType = promoteTypes(highType, maybe_dtype);
+  }
+
+  return Compute(
+      "aten_stack", outputShape, [&](const std::vector<VarHandle>& axes) {
+        if (hasEmptyInputs) {
+          return ExprHandle(0);
+        }
+
+        int64_t dim_ = c10::get<int64_t>(argDim);
+        auto dim = normalizeAndCheckIndex(dim_, axes.size());
+        auto inputBuf = inputList[0];
+        std::vector<ExprHandle> newAxes(axes.begin(), axes.begin() + dim);
+        newAxes.insert(newAxes.end(), axes.begin() + dim + 1, axes.end());
```

Comment:
This doesn't look right. First line makes
```
newAxes[:] = axes[:]
```
Second line then appends it with `axes[dim+1:]`.

That means that if we have 10-d tensors and dim=0 the stacked tensor will have rank 19. I think it should be 11-d instead, am I missing something?

Also, please update the comment to describe what new axis are created by this op and where they are inserted.

Reply:
The first line makes
```
newAxes = axes[:dim]
```
Overall, there will be N-1 elements in newAxes where N is the size of axes.

Reply:
Ah, got it, it make sense. Maybe it'd be easier to grasp if we call `newAxes` as `outputAxes` and `axes` as `inputAxes` - currently "new" corresponds to inputs which is somewhat counterintuitive (even though I understand why it's done that way - we're deducing input indexes from output indexes).